packages/@dcl/sdk-commands/src/commands/deploy/index.ts

```diff
@@ -0,0 +1,208 @@
+
+  if (options.args['--target'] && options.args['--target-content']) {
+    throw new CliError(`You can't set both the 'target' and 'target-content' arguments.`)
+  }
+
+  // Obtain list of files to deploy
+  const files = await getFiles(options.components, dir)
+
+  validateFilesSizes(files)
+
+  const contentFiles = new Map(files.map((file) => [file.path, file.content]))
+  const sceneJson = await getValidSceneJson(options.components, dir)
+
+  const { entityId, files: entityFiles } = await DeploymentBuilder.buildEntity({
+    type: EntityType.SCENE,
+    pointers: sceneJson.scene.parcels,
+    files: contentFiles,
+    metadata: sceneJson
+  })
+
+  // Signing message
+  const messageToSign = entityId
+  const { signature, address, chainId } = await getAddressAndSignature(
+    options.components,
+    messageToSign,
+    sceneJson,
+    files,
+    {
+      openBrowser,
+      linkerPort,
+      isHttps: !!options.args['--https'],
+      skipValidations:
+        !!options.args['--skip-validations'] || !!options.args['--target'] || !!options.args['--target-content']
+    }
+  )
+  const authChain = Authenticator.createSimpleAuthChain(entityId, address, signature)
+
+  // Uploading data
+  const catalyst = await getCatalyst(options.args['--target'], options.args['--target-content'])
+
+  printProgressInfo(options.components.logger, `Uploading data to: ${catalyst.getContentUrl()}...`)
+
+  const deployData = { entityId, files: entityFiles, authChain }
+  const position = sceneJson.scene.base
+  const network = chainId === ChainId.ETHEREUM_GOERLI ? 'goerli' : 'mainnet'
+  const sceneUrl = `https://play.decentraland.org/?NETWORK=${network}&position=${position}`
+
+  try {
+    const response = (await catalyst.deploy(deployData, {
+      timeout: '10m'
+    })) as { message?: string }
+
+    if (response.message) {
+      printProgressInfo(options.components.logger, response.message)
+    }
+    printSuccess(options.components.logger, 'Content uploaded', sceneUrl)
+  } catch (e: any) {
+    error('Could not upload content:')
+    console.log(e)
+  }
+}
+
+async function getCatalyst(target?: string, targetContent?: string) {
+  if (target) {
+    return new CatalystClient({ catalystUrl: target.endsWith('/') ? target.slice(0, -1) : target })
+  }
+
+  if (targetContent) {
+    return new ContentClient({ contentUrl: targetContent })
+  }
+
+  return CatalystClient.connectedToCatalystIn({ network: 'mainnet' })
+}
+
+interface LinkOptions {
+  openBrowser: boolean
+  linkerPort?: number
+  isHttps: boolean
+  skipValidations: boolean
+}
+
+const runLinkerApp = async (
+  components: CliComponents,
+  rootCID: string,
+  scene: Scene,
+  files: IFile[],
+  options: LinkOptions
+): Promise<LinkerResponse> => {
+  const { linkerPort, ...opts } = options
+  try {
+    const res = await link(components, scene, files, linkerPort!, rootCID, opts)
+    return res
+  } catch (e) {
+    throw new CliError(`Error running linking app: ${e}`)
+  }
+}
```

Comment:
can this logic be moved into link? It doesn't make much sense to have this wrapping function.

Also I'd find a more expressive name for `link` function. It is hard to tell what it does

Reply:
just kept it for consistency with the old CLI, I think we don't even need to move it to the `link` fn, we can just remove it completely and expect errors to bubble up and it will be the same